### ISSUE TITLE
"Running Run Apex Tests" message/command still running even though test run finished

### ISSUE THREAD

**Issue by marcos-toranzo** [forcedotcom/salesforcedx-vscode#2981]
### Summary

When I am running a unit test, the message "Running Run Apex test(s)" message keeps showing. It looks like the listener on my end is not registering that a test run is completed. This is an issue with running tests through my sandbox. This is not the usual behavior, but it happens many times in the day. Because the notification doesn't allow me to close it, I'm forced to restart the VSCode in order to get rid of it, nothing else seems to work.

### Steps To Reproduce:

1. Open a project and use Sandbox to deploy and run tests from
2. Make changes
3. Either select "Run Test" via code lens or "Run Single Test" from the tests tab.

### Expected result

The CLI should recognize that a unit test has finished running and display results in output.

### Actual result

The CLI is stuck running specified test doesn't seem to know that test has actually finished running.

### Additional information

![Annotation 2021-02-24 163411](https://user-images.githubusercontent.com/68938936/109068739-421d1c80-76be-11eb-9734-d3602bdfb310.png)

**VS Code Version**: 1.53.2

**SFDX CLI Version**: 7.85.1

**OS and version**: Windows 10 20H2

**Comment by lcampos** [comment]
Hi @marcos-toranzo , we released several fixes around the code for running Apex tests (version 51.2.0 of the extensions). Can you confirm if these fixes address the issues you are seeing ? I'm not able to replicate this behavior on my end.

**Comment by marcos-toranzo** [comment]
Hi @lcampos . Thanks for the quick reply. I updated the extensions to version 51.2.0 this morning and I'm still facing the same issue. 1 out of 6 tests I've ran today got stuck in "Running Run Apex Tests". I checked the test finished on my Developer Console. It appears to happen randomly. When there is some of this messages stuck, I can run the same test (one of those that got stuck) and it could run correctly, or not. The same happens if I try to run a test that is not one that got stuck: it could run correctly, or not. What bothers me the most is the fact that the notification/message can not be dismissed (or at least I don't know how), so I'm forced to restart VSCode in order to get rid of them. Thanks in advance.
